RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+# RFC: Amplify JS TypeScript Improvements
+
+Amplify JS is looking to improve our TypeScript support across the library to better serve our customers and provide a more intuitive & idiomatic developer experience. To this end, we are requesting your feedback on a variety of changes and improvements that will be available to you in our next major version release.
+
+This RFC is broken down into sections covering:
+
+- Library-wide TypeScript improvements
+- Changes to some of our core utilities such as `Hub` and Amplify configuration
+- Specific improvements to the `Auth`, `Storage`, and `API` categories and associated APIs
+
```

Comment:
As a reader, I care more of the `Auth`, `Storage` and `API` over the TypeScript improvement or other changes. I think we should reorder these.

Reply:
I agree with that ^

Reply:
Yeah that's fair, I wanted them to match the ordering in the doc & library-wide improvements seemed like a good "intro" section but we can switch it around

Reply:
I kind of disagree. I like to see the bird's eye view before drilling in